crates/meilisearch/src/routes/indexes/fields.rs

```diff
@@ -0,0 +1,328 @@
+    });
+
+    let sort_by = facet_sort_config
+        .get(&field_name)
+        .or_else(|| facet_sort_config.get("*"))
+        .map(|s| match s {
+            FacetValuesSort::Alpha => "alpha",
+            FacetValuesSort::Count => "count",
+        })
+        .unwrap_or("alpha")
+        .to_string();
+
+    let mut locales = Vec::new();
+    for rule in localized_rules {
+        if rule.match_str(&field_name) == milli::PatternMatch::Match {
+            locales.extend(rule.locales().iter().map(|l| {
+                let loc: Locale = (*l).into();
+                format!("{:?}", loc).to_lowercase()
+            }));
+        }
+    }
+
+    Field {
+        name: field_name,
+        displayed: FieldDisplayConfig { enabled: displayed_enabled },
+        searchable: FieldSearchConfig { enabled: searchable_enabled },
+        distinct: FieldDistinctConfig { enabled: distinct_enabled },
+        filterable: FieldFilterableConfig {
+            enabled: is_filterable,
+            facet_search: FieldFacetSearchConfig { sort_by },
+            filter: FieldFilterConfig { equality: is_filterable, comparison: is_filterable },
+        },
+        localized: FieldLocalizedConfig { locales },
+    }
+}
+
+pub async fn get_index_fields(
+    index_scheduler: GuardedData<ActionPolicy<{ actions::INDEXES_GET }>, Data<IndexScheduler>>,
+    index_uid: web::Path<String>,
+    params: AwebQueryParameter<ListFields, DeserrQueryParamError>,
+) -> Result<HttpResponse, ResponseError> {
+    let index_uid = meilisearch_types::index_uid::IndexUid::try_from(index_uid.into_inner())?;
+    let index = index_scheduler.index(&index_uid)?;
+    let rtxn = index.read_txn()?;
+
+    let all_field_names: Vec<String> =
+        index.fields_ids_map(&rtxn)?.names().map(|s| s.to_string()).collect();
+
+    let settings = settings(&index, &rtxn, SecretPolicy::RevealSecrets)?;
+
+    let displayed_fields = match &*settings.displayed_attributes {
+        Setting::Set(f) => f.clone(),
+        Setting::Reset | Setting::NotSet => vec!["*".to_string()],
+    };
+    let searchable_fields = match &*settings.searchable_attributes {
+        Setting::Set(f) => f.clone(),
+        Setting::Reset | Setting::NotSet => vec!["*".to_string()],
+    };
+    let distinct_field = match &settings.distinct_attribute {
+        Setting::Set(f) => Some(f),
+        _ => None,
+    };
+    let filterable_rules = match &settings.filterable_attributes {
+        Setting::Set(r) => r.clone(),
+        _ => Vec::new(),
+    };
+    let localized_rules: Vec<LocalizedAttributesRule> = match &settings.localized_attributes {
+        Setting::Set(r) => r.iter().cloned().map(|r| r.into()).collect(),
+        _ => Vec::new(),
+    };
+    let facet_sort_config = match &settings.faceting {
+        Setting::Set(f) => match &f.sort_facet_values_by {
+            Setting::Set(c) => c.clone(),
+            _ => BTreeMap::new(),
+        },
+        _ => BTreeMap::new(),
+    };
```

Comment:
This doesn't seem reliable. It's just guessing which fields are what, but the actual behavior in the engine may differ from this logic.

The data actually exists and can be retrieved easily by using `index.fields_ids_map_with_metadata` instead of `index.fields_ids_map`. There should be everything in it [Metadata](https://github.com/meilisearch/meilisearch/blob/c37ed05f49e0b091794e6e951d20638926bafddc/crates/milli/src/fields_ids_map/metadata.rs#L16) and if anything is missing, there must be a way to access it directly from the index scheduler